benchmark/bench_string_utils.cpp

```diff
@@ -97,3 +97,42 @@ static void BMBoostStartsWithFalseCase(benchmark::State &state) {
 }
 
 BENCHMARK(BMBoostStartsWithFalseCase);
+
+static void BMTrimLeft(benchmark::State &state) {
+  std::string kInputString = "  1234 ";
```

Comment:
I think you need to construct the string inside the benchmark loop, since the trimming mutates it in place - so after the first iteration, each subsequent trim call won't need to trim anything anymore.

We are going to also measure the string construction/destruction in that case, but assuming small string optimization, the time for that should be constant, so the comparison with the Boost implementation would still be valid I believe.

Reply:
dang, how did I miss that?!

fixed now, plus, used benchmark's API to not time the string creation (dtor may still be timed tho)